### PR TITLE
Merge DB connection-level and call-level attributes

### PR DESCRIPTION
**Pull request by lmolkova**
## Changes

Merges connection and call-level attributes for databases.

DB instrumentations in the general case are operating on the logical/public surface level and not always aware of connections.
Separation into connection and call level is also confusing. 

Extracted from #768, related to #690

## Merge requirement checklist

* [x] [CONTRIBUTING.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md) guidelines followed.
* [x] Change log entry added, according to the guidelines in [When to add a changelog entry](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md#when-to-add-a-changelog-entry).
  * If your PR does not need a change log, start the PR title with `[chore]`
* ~~[schema-next.yaml](https://github.com/open-telemetry/semantic-conventions/blob/main/schema-next.yaml) updated with changes to existing conventions.~~

### CODE REVIEW

File: model/trace/database.yaml [hunk 1]
```diff
@@ -6,95 +6,82 @@ groups:
...
     extends: db
     brief: >
-      Connection-level attributes for Microsoft SQL Server
+      Attributes for Microsoft SQL Server
     attributes:
       - ref: db.mssql.instance_name
-        tag: connection-level-tech-specific
+        tag: tech-specific
+      - ref: db.jdbc.driver_classname
+        tag: tech-specific
```
Comment (trask) [inline model/trace/database.yaml:48]: can mssql extend sql?

File: model/trace/database.yaml [hunk 2]
```diff
@@ -141,79 +128,81 @@ groups:
...
     extends: 'db'
     brief: >
-      Call-level attributes for SQL databases
+      Attributes for SQL databases
     attributes:
       - ref: db.sql.table
-        tag: call-level-tech-specific
+        tag: tech-specific
+      - ref: db.jdbc.driver_classname
+        tag: tech-specific
```
Comment (trask) [inline model/trace/database.yaml:191]: I'd support removing this attribute in a follow-up (we are not capturing it in Java)

File: model/trace/database.yaml [hunk 3]
```diff
@@ -228,36 +217,26 @@ groups:
...
-
-  - id: db.tech
-    type: span
-    brief: "Semantic convention group for specific technologies"
-    constraints:
-      - include: 'db.cassandra'
-      - include: 'db.redis'
-      - include: 'db.mongodb'
-      - include: 'db.sql'
-      - include: 'db.cosmosdb'
```
Comment (trask) [inline model/trace/database.yaml:263]: I didn't follow where this was being used before?

File: model/trace/database.yaml (hunk 2 above)
Comment (lmolkova) [reply model/trace/database.yaml:189]: created https://github.com/open-telemetry/semantic-conventions/issues/782

File: model/trace/messaging.yaml [hunk 4]
```diff
@@ -82,15 +82,11 @@ groups:
       - ref: messaging.message.body.size
       - ref: server.address
       - ref: network.peer.address
-        tag: connection-level
```
Comment (jack-berg) [inline model/trace/messaging.yaml:85]: Are these changes to messaging semconv intended to be part of this PR?

Comment (jack-berg) [reply model/trace/messaging.yaml:85]: Same question applies to the changes in `messaging-common.yaml`

Comment (lmolkova) [reply model/trace/messaging.yaml:85]: this is a dead code in messaging semconv which was copied over from DB. 

I'd rather keep it, but can remove this change if you feel strong about it.

**Review by jack-berg — APPROVED** [review]

**Review by joaopgrassi — APPROVED** [review]

**Review by alanwest — APPROVED** [review]

**Pull request merged**